website/project/model.py

```diff
@@ -2343,10 +2347,12 @@ def add_contributors(self, contributors, auth=None, log=True, save=False):
         :param save: Save after adding contributor
         """
         for contrib in contributors:
-            self.add_contributor(
+            added = self.add_contributor(
                 contributor=contrib['user'], permissions=contrib['permissions'],
                 visible=contrib['visible'], auth=auth, log=False, save=False,
             )
+            if added:
+                project_signals.contributor_added.send(self, contributor=contrib['user'])
```

Comment:
This gets tricky because there are many places where `add_contributor` is called where we do not want to trigger the associated event (for example, when registrations are created).